### ISSUE TITLE
Missing body using DELETE and ProxyManager

### ISSUE THREAD

**Issue by Lejurn** [urllib3/urllib3#1680]
Hi,

I discovered a strange issue in urllib3 version 1.24.1 and python3 version 3.7.3 . The body of the DELETE requests is missing, using a proxy and a HTTPS connection. Event though the content-length is correct. I'm using BurpSuite as proxy. 

First I discovered the issue using the requests lib, but was able to reproduce the issue using urllib3:

```python
import urllib3
proxy = urllib3.ProxyManager('https://localhost:8080/', cert_reqs='CERT_NONE', assert_hostname=False)
proxy.request('DELETE', 'https://192.168.5.1/test', body="test")
```
Using the request lib I was able to trace the request, through urllib3/contrib/pyopenssl.py, to the OpenSSL.SSL.Connection.sendall function, the body was still intact.
I'm not sure why the body is missing, because these function just seems to write the buffer to the corresponding socket. 

Using http.client I'm able to send DELETE requests including the body, so it don't seems to be an issue related to the proxy software.

**Comment by sethmlarson** [comment]
Do you have a pcap from wireshark? Might help debugging in this case.

**Comment by Lejurn** [comment]
I attached the pcapng file (zip) and sslkeylog from the local interface.
As far as I know, I cannot export a sslkeylog file using burp, so the traffic after the proxy is pretty much useless. 

I recorded a POST and DELETE request, just for comparison. Within burp the request body is missing. But the captured http traffic looks fine for me. Maybe it is a proxy issue after all.

[urllib3ssllog.txt](https://github.com/urllib3/urllib3/files/3634938/urllib3ssllog.txt)
[urllib3.zip](https://github.com/urllib3/urllib3/files/3634942/urllib3.zip)
